### PR TITLE
Support updating classification when a document's project context changes.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/5343

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs [hunk 1]
```diff
@@ -503,10 +507,11 @@ private void ParseIfThisDocument(Solution oldSolution, Solution newSolution, Doc
             {
                 if (_workspace != null)
                 {
-                    var openDocumentId = _workspace.GetDocumentIdInCurrentContext(_subjectBuffer.AsTextContainer());
-                    if (openDocumentId == documentId)
+                    var relatedDocuments = _workspace.GetRelatedDocumentIds(_subjectBuffer.AsTextContainer());
+                    if (relatedDocuments.Contains(documentId))
                     {
-                        EnqueueParseSnapshotTask(newSolution.GetDocument(documentId));
+                        var documentIdInCurrentContext = _workspace.GetDocumentIdInCurrentContext(documentId);
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs]: ```documentId``` should be idempotent under ```GetDocumentIdInCurrentContext``` in this case, right?

**Comment by dpoeschl** [comment]
While this change generally seems sound, I'm not convinced it will fix the problem in https://github.com/dotnet/roslyn/issues/5343. Looking at it today, it seemed that we were correctly notifying the shared hierarchy of the context switch, but we were not then correctly responding to their notification of the same event, so the Roslyn idea of "current context" went out of sync with that of the shared hierarchy. This could be observed in the repro steps by clicking in the editor after the attempted context change, and observing the NavBar switch back to the previous context. This appears to be a regression from Dev14 RTM, and could certainly cause the classification issue you're trying to fix here. Let me investigate that part a bit before we consider taking this change.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/dbg/eta please

**Comment by CyrusNajmabadi** [comment]
retest prtest/mac/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/win/dbg/eta  please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Pull request closed without merging**